### ISSUE TITLE
How can we make the Terraform plan step faster?

### ISSUE THREAD

**Issue by alikhajeh1** [infracost/infracost#1073]
Sometimes running the Infracost CLI can feel slow as the TF Plan step is usually the slowest step. So far we've done a few things to help speed-up the CLI (lookup prices in parallel, cache TF init, [cache the TF plan](https://github.com/infracost/infracost/pull/1068) in some cases). 

The aim of this issue is to see what else we can do about the TF plan step, for example:
- Running TF plan in parallel for multi-project runs? We currently run them in serial for mono repos and other multi-project setups.
- https://www.terraform.io/docs/cli/commands/apply.html#parallelism-n

**Comment by aliscott** [comment]
I've added PR for paralellizing multi-project runs https://github.com/infracost/infracost/pull/1193. 
I've tested this against our infra which uses a config file two projects, similar to:

```
version: 0.1
projects:
  - path: dev
    usage_file: dev/infracost-usage.yml
  - path: prod
    usage_file: prod/infracost-usage.yml
```

Here's the results from my machine 
```
MacBook Pro (13-inch, 2020, Four Thunderbolt 3 ports)
Processor 2 GHz Quad-Core Intel Core i5
Memory 16 GB 3733 MHz LPDDR4X
```

| Code | Options | Avg time (3 runs) |
|-|-|-|
| master | - | 73.4s |
| master | terraform_plan_flags: "-parallelism=20" | 85.9s |
| master | terraform_plan_flags: "-parallelism=40" | 63.2s |
| [PR 1193](https://github.com/infracost/infracost/pull/1193) | - | 40.9s |
| [PR 1193](https://github.com/infracost/infracost/pull/1193) | terraform_plan_flags: "-parallelism=20" | 33.0s |
| [PR 1193](https://github.com/infracost/infracost/pull/1193) | terraform_plan_flags: "-parallelism=40" | 33.8s |

So it seems that parallelising the multi-project runs has a big effect. Adjusting the Terraform parallelism might also have an effect to an extent, but this seems less clear from the results.

**Comment by aliscott** [comment]
#1193 is now released in v0.9.15, so we default to running multi-project runs in parallel. Closing this issue just now, but we can reopen if we have further ideas for improvements.

**Comment by aliscott** [comment]
I've disabled this by default in CI at the moment until v0.9.16 is released due to some race conditions. These have been fixed in https://github.com/infracost/infracost/pull/1206.
